src/video_core/gpu.h

```diff
@@ -164,6 +168,26 @@ class GPU {
     /// Returns a reference to the GPU DMA pusher.
     Tegra::DmaPusher& DmaPusher();
 
+    void IncrementSyncPoint(const u32 syncpoint_id);
+
+    u32 GetSyncpointValue(const u32 syncpoint_id) const;
+
+    void RegisterSyncptInterrupt(const u32 syncpoint_id, const u32 value);
+
+    bool CancelSyncptInterrupt(const u32 syncpoint_id, const u32 value);
+
+    void Guard(bool guard_set) {
```

Comment:
This should just be:

```suggestion
    std::unique_lock<std::mutex> LockSyncMutex() {
      return std::unique_lock{sync_mutex};
    }
```

Which allows for the mutex to automatically be unlocked in a safe manner. This also allows unlocking/relocking the mutex if it's necessary.